iothub/device/src/Exceptions/IotHubClientException.cs

```diff
@@ -211,5 +176,31 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
             info.AddValue(IsTransientValueSerializationStoreName, IsTransient);
             info.AddValue(TrackingIdValueSerializationStoreName, TrackingId);
         }
+
+        private Dictionary<IotHubErrorCode, bool> mapping = new Dictionary<IotHubErrorCode, bool>
```

Comment:
add a mapping between the error code and IsTrasient flag

Reply:
Since it is a boolean, you could simply have a hashset of codes that are transient.